src/types.ts

```diff
@@ -101,12 +101,17 @@ export type Blockchain =
       ens?: NamingServiceSource;
       zns?: NamingServiceSource;
       cns?: NamingServiceSource;
+      web3Provider?: any;
```

Comment:
could be `Provider` too?

Reply:
When I did that and tried to used the ethers.js provider IDE started to split a bunch of errors. So I left that as any and sending it further as my Provider. 
The Biggest issue so far is that I don't understand where can I get the providers from to test. The one I am getting from ethers is obviously wrong, it doesn't have sendAssync function which should be implemented in each provider